### PR TITLE
Implement Revoke Certificates Command

### LINKED ISSUE #87: Revoke Certificate
# 🗣 Context

Revoke a lost, stolen, compromised, or expiring signing certificate.

# 💬 Narrative

As a user
I want to revoke a compromised certificate
So that so that it can no longer be used to sign apps

# 📝 Notes

Covering this [API](https://developer.apple.com/documentation/appstoreconnectapi/revoke_a_certificate).

# 🎨 Design

```
asc certificates revoke XCYZ ABCD EFGH
```

Output:
```
🚮 Certificate `XCYZ` revoked.
🚮 Certificate `ABCD ` revoked.
🚮 Certificate `EFGH ` revoked.
```

# ✅ Acceptance Criteria

**GIVEN** I provide a valid input
**WHEN** I issue a revoke command
**THEN** the certificates I specified are revoked.

**GIVEN** I provide **some** invalid input (eg, one of the serials I provide is invalid)
**WHEN** I issue a revoke command
**THEN** then NO certificates are revoked and I receive an appropriate error message.

### PR DESCRIPTION
**Pull request by DechengMa**
Implement Revoke Certificate Command, Closes #87 
 
# 📝 Summary of Changes

Changes proposed in this pull request:

- Implement RevokeCertificatesCommand

# 🧐🗒 Reviewer Notes

## 💁 Example

Usage `asc certificates revoke [<serials> ...]`

Example Output: 
```
🚮 Certificate `12345ABCDE` revoked.
🚮 Certificate `54321ZXCVB` revoked.
```

## 🔨 How To Test

`swift run certificates revoke  12345ABCDE 54321ZXCVB`

### CODE REVIEW

File: Sources/AppStoreConnectCLI/Commands/Certificates/RevokeCertificatesCommand.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            throw ValidationError("Invalid input, you must provide at least one certificate serial number")
+        }
+    }
+
+    func run() throws {
+        let service = try makeService()
+        _ = try service.revokeCertificates(serials: serials)
+
+        serials.forEach {
+            print("🚮 Certificate `\($0)` revoked.")
```
Comment (adamjcampbell) [inline Sources/AppStoreConnectCLI/Commands/Certificates/RevokeCertificatesCommand.swift:29]: interesting, a different kind of output for this command

File: Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift [hunk 2]
```diff
@@ -26,13 +26,30 @@ class AppStoreConnectService {
...
+            .await()
+
+        return Certificate(sdkCertificate)
     }
 
     func createCertificate(with options: CreateCertificateOptions) -> AnyPublisher<Certificate, Error> {
         CreateCertificateOperation(options: options).execute(with: requestor)
     }
 
+    func revokeCertificates(serials: [String]) throws -> [Void] {
```
Comment (adamjcampbell) [inline Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift:40]: 🤔 I don't think the void array output is useful:

```suggestion
    func revokeCertificates(serials: [String]) throws {
```

File: Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift [hunk 3]
```diff
@@ -26,13 +26,30 @@ class AppStoreConnectService {
...
+        let certificatesIds = try serials.map {
+            try ReadCertificateOperation(options: .init(serial: $0))
+                .execute(with: requestor)
+                .await()
+                .id
+            }
+
+        return try RevokeCertificatesOperation(options: .init(ids: certificatesIds))
```
Comment (adamjcampbell) [inline Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift:48]: To match the new method signature:

```suggestion
        try RevokeCertificatesOperation(options: .init(ids: certificatesIds))
```

File: Sources/AppStoreConnectCLI/Services/Operations/RevokeCertificateOperation.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    init(options: Options) {
+        self.options = options
+    }
+
+    func execute(with requestor: EndpointRequestor) throws -> AnyPublisher<Void, Swift.Error> {
+        let requests = options.ids.compactMap {
+                requestor
+                    .request(APIEndpoint.revokeCertificate(withId: $0))
+                    .eraseToAnyPublisher()
+            }
```
Comment (adamjcampbell) [inline Sources/AppStoreConnectCLI/Services/Operations/RevokeCertificateOperation.swift:24]:
```suggestion
        let requests = options.ids.map { id in
            requestor.request(APIEndpoint.revokeCertificate(withId: id))
        }
```

File: Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift (hunk 2 above)
Comment (DechengMa) [reply Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift:40]: @adamjcampbell will get `Unexpected non-void return value in void function` if remove `[Void]`

File: Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift [hunk 5]
```diff
@@ -26,13 +26,30 @@ class AppStoreConnectService {
...
         CreateCertificateOperation(options: options).execute(with: requestor)
     }
 
+    func revokeCertificates(serials: [String]) throws {
+        let certificatesIds = try serials.map {
+            try ReadCertificateOperation(options: .init(serial: $0))
+                .execute(with: requestor)
+                .await()
+                .id
+            }
```
Comment (adamjcampbell) [inline Sources/AppStoreConnectCLI/Services/AppStoreConnectService.swift:46]: The closing brace should match the indentation level of the line that has the opening brace
```suggestion
        }
```

**Pull request merged**